dask_cuda/benchmarks/local_cudf_merge.py

```diff
@@ -368,9 +375,27 @@ def parse_args():
 
 
 if __name__ == "__main__":
+    args = parse_args()
+
+    # Raise error early if "explicit-comms" is not allowed
+    if (
+        args.backend == "explicit-comms"
+        and dask.config.get(
+            "dataframe.query-planning",
+            None,
+        )
+        is not False
+    ):
+        raise NotImplementedError(
+            "The 'explicit-comms' config is not yet supported when "
+            "query-planning is enabled in dask. Please use the legacy "
+            "dask-dataframe API (set the 'dataframe.query-planning' "
+            "config to `False` before executing).",
+        )
```

Comment:
It seems you're not addressing this for `local_cudf_groupby` and `local_cudf_merge`, are explicit-comms still supported for those? If not, I'd suggest moving this into [`parse_benchmark_args`](https://github.com/rapidsai/dask-cuda/blob/127de5956ffd709e97d1fdda8fba71541843b4fc/dask_cuda/benchmarks/utils.py#L317-L320) just before the function returns, and as such you'd cover all three cases.

Reply:
Right - sorry. I'll use the same message for `local_cudf_groupby`. This is `local_cudf_merge`, so I assume you are also asking why I am not checking for this in `local_cudf_shuffle`? If so, it's because that benchmark does not use the dask config (it uses `dask_cuda.explicit_comms.dataframe.shuffle.shuffle` directly).

Reply:
> Right - sorry. I'll use the same message for `local_cudf_groupby`.

I'd still suggest simply moving this to `parse_benchmarks_args`, this will cover all benchmarks and avoid code duplication, which is essentially why we wrote it that way to begin with.

> This is `local_cudf_merge`, so I assume you are also asking why I am not checking for this in `local_cudf_shuffle`?

Yes, sorry, that is what I meant. 😅 

> If so, it's because that benchmark does not use the dask config (it uses `dask_cuda.explicit_comms.dataframe.shuffle.shuffle` directly).

But would using it directly still work with query-planning enabled? If not then moving the exception above to `parse_benchmarks_args` would also cover it and avoid more boilerplate code.

Reply:
>I'd still suggest simply moving this to parse_benchmarks_args, this will cover all benchmarks and avoid code duplication, which is essentially why we wrote it that way to begin with.

Ah - I see what you mean. Using `dask_cuda.explicit_comms.dataframe.shuffle.shuffle` directly works fine, so it would still be good to allow that option for the shuffle benchmark somehow.

Reply:
Ok, then how about still moving this to `parse_benchmarks_args` but adding another kwarg to it that would still allow running select benchmarks (e.g., `local_cudf_shuffle`) with planning-query enabled. Something like below:

```python
def parse_benchmark_args(description="Generic dask-cuda Benchmark", args_list=[], *, force_explicit_comms=False):
...

    if args.multi_node and len(args.hosts.split(",")) < 2:
        raise ValueError("--multi-node requires at least 2 hosts")

    # Raise error early if "explicit-comms" is not allowed
    if (
        args.backend == "explicit-comms"
        and dask.config.get(
            "dataframe.query-planning",
            None,
        )
        is not False
        and force_explicit_comms is False
    ):
        raise NotImplementedError(
            "The 'explicit-comms' config is not yet supported when "
            "query-planning is enabled in dask. Please use the legacy "
            "dask-dataframe API (set the 'dataframe.query-planning' "
            "config to `False` before executing).",
        )

    return args
```

Then `local_cudf_{groupby,merge}` remain as they were and for `local_cudf_shuffle` you simply add `force_explicit_comms=True` to [this line](https://github.com/rapidsai/dask-cuda/blob/127de5956ffd709e97d1fdda8fba71541843b4fc/dask_cuda/benchmarks/local_cudf_shuffle.py#L257).

WDYT?

Reply:
Okay, sounds good. Pushed something very similar to your solution just now.